vm/resolutions.go

```diff
@@ -255,50 +264,15 @@ func (vm *VM) Accepted(ctx context.Context, b *StatefulBlock) {
 	delete(vm.verifiedBlocks, b.ID())
 	vm.verifiedL.Unlock()
 
-	// Update replay protection heap
-	//
-	// Transactions are added to [seen] with their [expiry], so we don't need to
-	// transform [blkTime] when calling [SetMin] here.
-	blkTime := b.Tmstmp
-
-	// Verify if emap is now sufficient (we need a consecutive run of blocks with
-	// timestamps of at least [ValidityWindow] for this to occur).
-	if !vm.IsReady() {
-		select {
-		case <-vm.seenValidityWindow:
-			// We could not be ready but seen a window of transactions if the state
-			// to sync is large (takes longer to fetch than [ValidityWindow]).
-		default:
-			seenValidityWindow, err := vm.syncer.Accept(ctx, b.ExecutionBlock)
-			if err != nil {
-				vm.Fatal("syncer failed to accept block", zap.Error(err))
-			}
-			if seenValidityWindow {
-				vm.seenValidityWindowOnce.Do(func() {
-					close(vm.seenValidityWindow)
-				})
-			}
+	for _, sub := range vm.acceptedSubscriptions {
+		if err := sub.Accept(b); err != nil {
```

Comment:
Should these events be taking in a context? It seems like we are inserting a number of `context.TODO()`s where we could just pass them in as expected.

Reply:
Updated